django_migration_linter/migration_linter.py

```diff
@@ -54,12 +61,37 @@ def __init__(self, project_path, **kwargs):
         self.nb_erroneous = 0
         self.nb_total = 0
 
+        # Initialise cache. Read from old, write to new to prune old entries.
+        self.old_cache = Cache(self.django_path, self.cache_path)
+        self.new_cache = Cache(self.django_path, self.cache_path)
+        if not self.no_cache:
+            self.old_cache.load()
+
     def lint_migration(self, app_name, migration_name):
         print('({0}, {1})... '.format(app_name, migration_name), end='')
         self.nb_total += 1
 
+        md5hash = self.old_cache.md5(app_name, migration_name)
+        if md5hash in self.old_cache:
+            if self.old_cache[md5hash]['result'] == 'IGNORE':
+                print('IGNORE (cached)')
+                self.nb_ignored += 1
+            elif self.old_cache[md5hash]['result'] == 'OK':
+                print('OK (cached)')
+                self.nb_valid += 1
+            else:
+                print('ERR (cached)')
+                self.nb_erroneous += 1
+
+            if 'errors' in self.old_cache[md5hash]:
+                self.print_errors(self.old_cache[md5hash]['errors'])
+
+            self.new_cache[md5hash] = self.old_cache[md5hash]
+            return
+
         if self.should_ignore_migration(app_name, migration_name):
             print('IGNORE')
+            self.new_cache[md5hash] = {'result': 'IGNORE'}
```

Comment:
I think this is a bug:
if I call the linter with `django_migration_linter my_project/ --ignore-name 0001_initial`, it will be ignored and cached.
And if I call just after that `django_migration_linter my_project/`, the migrations with the name `0001_initial` will be ignored => KO